webSockets/receiver/receiver_test.go

```diff
@@ -0,0 +1,150 @@
+package receiver
+
+import (
+	"errors"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/mock"
+	"github.com/multiversx/mx-chain-core-go/data/typeConverters/uint64ByteSlice"
+	"github.com/multiversx/mx-chain-core-go/testscommon"
+	"github.com/multiversx/mx-chain-core-go/webSockets"
+	"github.com/multiversx/mx-chain-core-go/webSockets/data"
+	"github.com/stretchr/testify/require"
+)
+
+func createArgs() ArgsReceiver {
+	payloadConverter, _ := webSockets.NewWebSocketPayloadConverter(uint64ByteSlice.NewBigEndianConverter())
+	return ArgsReceiver{
+		BlockingAckOnError: false,
+		PayloadConverter:   payloadConverter,
+		Log:                &mock.LoggerMock{},
+		RetryDurationInSec: 1,
+	}
+}
+
+func TestNewReceiver(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
+		args := createArgs()
+		ws, err := NewReceiver(args)
+		require.NotNil(t, ws)
+		require.Nil(t, err)
+	})
+
+	t.Run("empty logger, should return error", func(t *testing.T) {
+		args := createArgs()
+		args.Log = nil
+		ws, err := NewReceiver(args)
+		require.Nil(t, ws)
+		require.Equal(t, err, core.ErrNilLogger)
+	})
+
+	t.Run("nil payload converter, should return error", func(t *testing.T) {
+		args := createArgs()
+		args.PayloadConverter = nil
+		ws, err := NewReceiver(args)
+		require.Nil(t, ws)
+		require.Equal(t, err, data.ErrNilPayloadConverter)
+	})
+
+	t.Run("zero retry duration in seconds, should return error", func(t *testing.T) {
+		args := createArgs()
+		args.RetryDurationInSec = 0
+		ws, err := NewReceiver(args)
+		require.Nil(t, ws)
+		require.Equal(t, err, data.ErrZeroValueRetryDuration)
```

Comment:
require should have: t, expected, actual
for the entire file

Reply:
done